sklego/model_selection.py

```diff
@@ -49,14 +61,17 @@ def __init__(
         self.train_duration = train_duration
         self.valid_duration = valid_duration
         self.gap_duration = gap_duration
+        self.n_splits = n_splits
+        self.maximise_training_set = False
 
     def join_date_and_x(self, X):
```

Comment:
This should have happened earlier, but maybe make this a hidden method? 

From

```
def join_date_and_x(self, X):
```

Into

```
def join_date_and_x_(self, X):
```

After all, the user should never touch it.

Reply:
leading underscore right, not trailing?